packages/teams-js/src/public/pages.ts

```diff
@@ -750,4 +750,94 @@ export namespace pages {
       return runtime.supports.pages ? (runtime.supports.pages.appButton ? true : false) : false;
     }
   }
+
+  /**
+   * Parameters for the NavigatewithinApp API
+   *
+   * @beta
+   */
+  export interface NavigateWithinAppParams {
```

Comment:
There will be still some clients which will be using this. We can't be sure that all of the clients are ported to newer parameters. Therefore, the interface will stay unless we are making some breaking changes in the future `(v3 or newer)`

Reply:
Once we make that breaking change, will we completely delete the interface?

Reply:
I think I am missing something. What is the purpose of deprecated tag if the properties won't be deleted, and if we do delete both the properties, we won't need empty interface either

Reply:
I am not holding your PR with my comment, but just curious to understand this style

Reply:
I agree with Lakhveer, it doesn't seem correct to mark these new fields as deprecated (it looks like you got these comments from the Context object. It would probably be more correct to remove the @deprecated tags from these but to still link to where the values come from so that app developers reading the documentation can figure out how to populate pageId and subPageId

Reply:
And actually, for subPageId I don't think it's that the app developer should take the value of subPageId from Context and pass it in here, it's that whatever value they pass to subPageId here will show up in their app after navigation in app.PageInfo.subPageId (used for passing data between pages)

Reply:
Resolved in latest commit